### PR TITLE
End-to-end runners and alignment generation

### PR DESCRIPTION
**Pull request by delucchi-cmu**
Implements 
- end-to-end runners to execute the partitioning pipeline in either a dask parallelization scheme, or without any parallelization
- method for reducing a raw histogram to the destination pixels
- method for printing out the metadata file expected by the current prototype catalog
- convenience methods for the above, that will be used in future pull requests as well.

### CODE REVIEW

File: src/partitioner/__main__.py [hunk 1]
```diff
@@ -1,9 +1,24 @@
...
 
+import dask_runner
+import single_runner
 from arguments import PartitionArguments
 
+
+def run():
+    """Pick a runtime environment and run the partitioner"""
+    match args.runtime:
+        case "single":
```
Comment (maxwest-uw) [inline src/partitioner/__main__.py:13]: should we/have you set "single" as the default case, and following that do we/should we have a set of default arguments for the partitioner?

File: src/partitioner/histogram.py [hunk 2]
```diff
@@ -60,3 +60,74 @@ def generate_partial_histogram(
...
+    for i in range(0, highest_order):
+        nested_sums.append(empty_histogram(i))
+    nested_sums.append(histogram)
+
+    # work backward - from highest order, fill in the sums of lower order pixels
+    for read_order in range(highest_order, 0, -1):
+        parent_order = read_order - 1
+        for index in range(0, len(nested_sums[read_order])):
+            parent_pixel = index >> 2
+            # print(f"{read_order} : {index} : {parent_order} : {parent_pixel}")
```
Comment (maxwest-uw) [inline src/partitioner/histogram.py]: do you want to keep this debug statement (?) around?

File: tests/partitioner/test_histogram.py [hunk 3]
```diff
@@ -51,3 +53,54 @@ def test_column_names():
     expected = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 8]
     npt.assert_array_equal(result, expected)
     assert (result == expected).all()
+
```
Comment (maxwest-uw) [inline tests/partitioner/test_histogram.py:56]: should probably have tests for the two value errors that can be raised (when histogram isn't right size and when it exceeds threshold)

**Review by maxwest-uw — COMMENTED** [review]
3 small things!

File: src/partitioner/__main__.py (hunk 1 above)
Comment (delucchi-cmu) [reply src/partitioner/__main__.py:13]: Yes - runtime is the default (arguments.py lines 29, 169, 217). I can add some documentation to the arguments module about defaults in a follow-up.

There are no single-execution-specific arguments, so there are no default arguments for that use case in particular, if you were asking about that.

File: src/partitioner/histogram.py (hunk 2 above)
Comment (delucchi-cmu) [reply src/partitioner/histogram.py]: No I do not!

File: tests/partitioner/test_histogram.py (hunk 3 above)
Comment (delucchi-cmu) [reply tests/partitioner/test_histogram.py:56]: Abso-toot-ly.

**Review by maxwest-uw — APPROVED** [review]

**Pull request merged**